Add support for custom headers

kumaranvram opened the pull request:
- This is prevent us from hard coding the header in the code.

- Now the headers can be sent as `--headers 'X-Auth-Token: i-am-a-token, X-Another-Dummy-Header: dummy-value'`
- Since we are allowing custom headers, the switch is changed to `--headers` instead of `--auth`

## Description

Fixes # (issue)

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have ran `deno task ci` to ensure that my code is formatted and linted.
- [ ] I have linked the related issue _(if there is no related issue please
      create one)_
- [ ] ~~I have added tests if applicable~~ (_needs tests to be ready_)

Commit:
Add support for custom auth headers

- This is prevent us from hard coding the header in the code.

- Now the headers can be sent as "--auth 'X-Auth-Token:
  i-am-a-token'"

postmodem commented on src/format.ts:
```diff
@@ -55,8 +56,8 @@ function queryToItem(
-        ...(authorization
-          ? [{ key: "Authorization", value: authorization }]
+        ...((authorization && authorizationHeader)
+          ? [{ key: authorizationHeader.trim(), value: authorization.trim() }]
```
This would set our auth key as the value for the generated requests right? We don't want that. Better to use an env variable and the person who downloads the collection can add it to their Postman. We can also have the generated README reflect this. Something like `._storefront_access_token` which is how we currently do it in the storefront-api-learning-kit.

kumaranvram replied:
For now, this PR would only set the blank values for the custom headers set in the postman collection. 

My 2 cents: I think we can mention this in the README to fill it up for now. We shall look at the enhancement of doing this once we complete the MVP. What do you think?

postmodem reviewed (changes requested):
see comment about the request header value

nidbhardwaj commented on src/cli.ts:
```diff
@@ -32,7 +32,9 @@ if (Deno.args.length < 1 || args.help || args.h) {
 const url = args._[0];
 let path = args.out;
-const authorization = args.auth;
+const authorizationString = args.auth;
```
can we accept a list instead of a single header to make it future-proof? Also, I think we should rename --auth to --headers as the argument. 
Something similar to the below code can we used.
```ts
const requestHeaders: Record<string, string> = {};
    headers.forEach(header => {
        const [key, value] = header.split(':');
        requestHeaders[key.trim()] = value.trim();
    });
    ```

nidbhardwaj replied:
Just saw your msg, I can pick this enhancement.

kumaranvram replied:
Done. Now the application accepts a list of headers under `--headers`

nidbhardwaj commented on src/cli.ts:
```diff
@@ -32,7 +32,9 @@ if (Deno.args.length < 1 || args.help || args.h) {
 let path = args.out;
-const authorization = args.auth;
+const authorizationString = args.auth;
+const authorizationHeader = authorizationString?.split(":")[0];
```
Can we avoid merging this PR since it is introducing a lot of changes that will be touched again with changing the input type to a list?

kumaranvram replied:
👍🏼

Commit:
Add support for adding custom headers

- Since we are allowing addition of custom headers, I am changing the --auth flag to --headers

postmodem reviewed (approved):
lgtm
